### ISSUE TITLE
Sick Chill plugin install fails on Stable Release channel [11.3-RELEASE]

### ISSUE THREAD

**Issue by vipero07** [ix-plugin-hub/iocage-plugin-index#107]
This may be related to sick chill's removal of python 27 support.

```
Install
Error: sickchill had a failure Exception: RuntimeError Message: Last 10 lines: /root/post_install.sh: git: not found /root/post_install.sh: pip: not found chown: /usr/local/app/sickchill: No such file or directory cp: /usr/local/app/sickchill/contrib/runscripts/init.freebsd: No such file or directory chmod: /usr/local/etc/rc.d/sickchill: No such file or directory Partial plugin destroyed
```

**Comment by fulder** [comment]
From what I can see there seems to be a fix for python upgrade for this which was merged to master just 3 days before this issue was created: https://github.com/ix-plugin-hub/iocage-plugin-index/pull/106. But this fix has only been merged back to the 12.1-RELEASE branch (at least following the PR history)

Although the error you are getting are probably because pip and git is not installed but the plugin has started to need them both after this PR: https://github.com/ix-plugin-hub/iocage-plugin-sickchill/pull/1.

So in order to fix this master could maybe be synced to 11.3-RELEASE if this is OK @kmoore134 ? Will running FreeNAS 11.3 support running plugins with newer FreeBSD versions or will this cause more issues?

Otherwise it could also by solved by creating a patch update for the 11.3-RELEASE branch only with the python update and git/pip installation (would probably be a good idea to lock the release branches in the SickChill plugin for different kind of releases instead of just having 1 master for all of them).

(A bit outside of this issue but I've tested installing the plugin on a 12.1-RELEASE TrueNAS Core system and it worked fine)

**Comment by turboture** [comment]
Hi
Is it possible for you to help us with a updated plugin or a patch?

**Comment by turboture** [comment]
@fulder 
Is it possible for you to help us?

**Comment by fulder** [comment]
Could this PR maybe solve the issue: https://github.com/ix-plugin-hub/iocage-plugin-index/pull/134 ?

I don't have a 11.3-RELEASE system to test with but if you could check you could probably try to install the plugin manually like this:

* `fetch https://raw.githubusercontent.com/fulder/iocage-plugin-index/11.3-RELEASE/sickchill.json`
* `iocage fetch -P sickchill.json --name sickchill_test dhcp=on`

**Comment by turboture** [comment]
@fulder  Thank you for trying to help us. I got these errors. I tried to install it from ssh and shell. Same result.

![image](https://user-images.githubusercontent.com/14950060/101922731-c0454900-3bce-11eb-8b70-e1fc6b37228b.png)

**Comment by turboture** [comment]
and now I cant delete the jail but that is a question for another forum.

**Comment by fulder** [comment]
Missed updating the `packagesite` from `pkg.cdn.trueos.org` to `pkg.FreeBSD.org` on my fork. Could you check if it works better after the latest push?

**Comment by fulder** [comment]
Have you tried destroying it with `iocage destroy <JAIL_NAME>` command? I've sometimes seen that my jails are destroyed correctly using the `iocage` command but seems to still be present using the old `jls` command. Only a reboot of my FreeNAS system would clean thoes up.

**Comment by turboture** [comment]
![image](https://user-images.githubusercontent.com/14950060/101927085-5c258380-3bd4-11eb-9cf8-bf47c2199160.png)

I can access sickchill despite the errors.

![image](https://user-images.githubusercontent.com/14950060/101927363-b0c8fe80-3bd4-11eb-922e-7f5a6ce3e8d6.png)


Tried the destroy command and it worked. Thanks.

**Comment by turboture** [comment]
Do I use your patch and change it to my jail? Or is it something I update/install from the plugin menu in freenas later?

**Comment by fulder** [comment]
One step closer then :) I'm guessing the plugin packages were installed correctly. The errors you're seeing are happening because of the SickChill requirements installed in the plugins `post_install.sh` script and one python pip dependency which cannot be installed.

I'm actually getting the exactly same error running TrueNAS-12.0-STABLE and inside a 12.1-RELEASE jail. I've created another PR for this second issue: https://github.com/ix-plugin-hub/iocage-plugin-index/pull/136 (Should probably be ported back to, at least 12.2-RELEASE and maybe also 11.3-RELEASE).

Just out of curiosity, wouldn't simply running an update on your jail solve this issue without any need for this patching? Have you tried doing that?

**Comment by turboture** [comment]
I have tried that. SickChill doesnt start after update. Wrong version of python perhaps?

**Comment by fulder** [comment]
>Do I use your patch and change it to my jail? Or is it something I update/install from the plugin menu in freenas later?

If the 11.3-RELEASE branch gets patched I think it would be enough to just refresh the plugin index from the UI and update SickChill from UI as well.

> I have tried that. SickChill doesnt start after update. Wrong version of python perhaps?

Sorry I got confused regarding all the release versions/branches. Double checking the [iocage code](https://github.com/iocage/iocage/blob/d04bd987f2a4efc541c03fa5557e41cd38108ddc/iocage_lib/cache.py#L22) and if I understand it correctly the index branch will be matched against the `freebsd-version` output FreeNAS/TrueNAS is based upon. So even updating your plugin will still lock it to the 11.3-RELEASE plugin branch, 11.3-RELEASE FreeBSD version for the plugin although as written in previous comment still use the `master` branch of the plugin while running the `post_install.sh` script. Furthermore the `post_install.sh` script in the plugin (even before the previously posted PR fix for the newest version) downloads the master version of the official SickChill repository. 

This is actually problematic for any kind of plugins which haven't got any release branchen but only one master and/or plugins which just fetches/downloades the latest version of the plugin code/binaries.

IMO the "right" fix for these issues would be to actually lock the versions of everything as much as possible in every step, so summarized for SickChill plugin:

* Index already locked to this repository branch 
* Plugin Jail version already locked in this repository for every plugin
* Plugin version **NOT** locked with release branches
* Downloaded plugin artifacts **NOT** locked to any version

But as the patch quickfix is in a PR we could probably just merge it and fix this SickChill issue this way.

**Comment by turboture** [comment]
Thank you for all your help.

**Comment by fulder** [comment]
@turboture the patch branch has now been merged to 11.3-RELEASE branch. If you get a free moment and want to test it I think it should work to just refresh your index and install the plugin from the UI.

**Comment by turboture** [comment]
Same error with libxm but that was to be expected.

This is a new error. The plugin starts at it should.

![image](https://user-images.githubusercontent.com/14950060/103295414-2f56d600-49f4-11eb-8310-474f6a02bab9.png)

On a sidenote. Is it possible to update an existing installation?

**Comment by fulder** [comment]
> Same error with libxm but that was to be expected.

Yes, like you say libxm error is still expected as https://github.com/ix-plugin-hub/iocage-plugin-index/pull/136 has also been merged but to master and need to be ported back to the other releases to work for them as well. I commented in the PR to merge it back to 11.3-RELEASE as well in order to solve this issue for you and others still running this version.

This missing lib seems to have a larger impact than I thought. Because it is a part of the SickChill requirements installation it also breaks all the other requirements. I.e. none of them are installed. The SickChill UI can still be reached and probably works good although e.g. parsing XML responses from different providers will probably not work. (Weird there weren't more issues regarding this, especially as this shouldn't work for the newest release either)

> This is a new error. The plugin starts at it should.

Couldn't see this while using 12.1-RELEASE of FreeBSD in my jail, the error actually means the system is running an older openssl version which accepts TLS versions below 1.3 :( this is problematic both while starting SickChill (as stated in the error message, for the providers hosted by/using Cloudflare) but also seems to break the SickChill requirements installation. I've tried to use the 11.3-RELEASE branch SickChill file and install it inside a 11.3 jail and could see this error as well. 11.3 is running `OpenSSL 1.0.2s-freebsd  28 May 2019` and we would need to upgrade it to at least 1.1.1 in order to make SickChill start up properly. Even though it is possible to install a newer version of OpenSSL from `pkg` the python is still compiled for this OS using the older version and requirement installation (of the cryptography python package) fails. 

This issue (and quickfix) is starting to grow much more than I first anticipated. Can I ask why you still run 11.3 version of FreeBSD at all? It is a version already marked as EOL (as of 30th of September this year) so it feels weird to try to hack in patches on FreeBSD level to make this plugin work. On the other hand locking the SickChill plugin version would make it unable to fetch any future updates which doesn't seem very optimal either. The most simple solution right now would be to just upgrade the jail to the newest FreeBSD version.

> On a sidenote. Is it possible to update an existing installation?

It should be possible to update an existing installation, all the packages are uninstalled and reinstalled using the newest manifest if I remember correctly.

**Comment by turboture** [comment]
To be absolutly honest. I didnt know 12.0 was stable. Still thought it was under development. Switching now.

**Comment by fulder** [comment]
@turboture did you manage to get some time to test if it works fine in the 12.X release? If so, and if it works, maybe we could close this issue?

**Comment by turboture** [comment]
Yes. Sorry for not reporting back. I installed as it should under 12.x.
